tools/themis/src/rules.rs

```diff
@@ -0,0 +1,351 @@
+pub fn author_is<'a>(expected: &'a str) -> impl Fn(&'a Workspace) -> Result<(), Error> {
+    move |workspace| {
+        let outliers = workspace
+            .members
+            .iter()
+            .filter(|pkg| !pkg.parsed.authors.iter().any(|author| author == expected))
+            .map(|pkg| PackageOutcome { pkg, value: Some(format!("{:?}", pkg.parsed.authors)) })
+            .collect::<Vec<_>>();
+
+        if !outliers.is_empty() {
+            return Err(Error::OutcomeError {
+                msg: "These packages need to be correctly authored".to_string(),
+                expected: Some(Expected { value: expected.to_string(), reason: None }),
+                outliers,
+            });
+        }
+
+        return Ok(());
+    }
+}
+
+/// ensure all non-private crates have a license
+pub fn publishable_has_license(workspace: &Workspace) -> Result<(), Error> {
+    let outliers = workspace
+        .members
+        .iter()
+        .filter(|pkg| {
+            utils::is_publishable(pkg)
+                && !(pkg.parsed.license.is_some() || pkg.parsed.license_file.is_some())
+        })
+        .map(|pkg| PackageOutcome { pkg, value: None })
+        .collect::<Vec<_>>();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These non-private packages should have a `license` specification".to_string(),
+            expected: None,
+            outliers,
+        });
+    }
+
+    return Ok(());
+}
+
+/// ensure all non-private crates have a license file
+///
+/// Checks for either one LICENSE file, or two LICENSE files, one of which
+/// is the Apache License 2.0 and the other is the MIT license.
+pub fn publishable_has_license_file(workspace: &Workspace) -> Result<(), Error> {
+    let outliers = workspace
+        .members
+        .iter()
+        .filter(|pkg| {
+            utils::pub_missing!(
+                pkg,
+                "LICENSE" || ("LICENSE-APACHE" && "LICENSE-MIT") || (pkg.parsed.license_file)?
+            )
+        })
+        .map(|pkg| PackageOutcome {
+            pkg,
+            value: pkg.parsed.license_file.as_ref().map(|l| l.to_string()),
+        })
+        .collect::<Vec<_>>();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These non-private packages should have a license file".to_string(),
+            expected: None,
+            outliers,
+        });
+    }
+
+    return Ok(());
+}
+
+/// ensure all non-private crates use the the same license
+pub fn publishable_has_unified_license(workspace: &Workspace) -> Result<(), Error> {
+    let mut license_groups = HashMap::new();
+
+    for pkg in &workspace.members {
+        if let Some(license) = &pkg.parsed.license {
+            *license_groups.entry(license).or_insert(0) += 1;
+        }
+    }
+
+    let (most_common_license, n_compliant) =
+        license_groups.into_iter().reduce(|a, b| if a.1 > b.1 { a } else { b }).unwrap();
```

Comment:
Why do we need to dynamicaly select most common license? Seems that the license we use is just a constant we know.

Reply:
Artifacts of, like you mentioned, a more generic design. I'll resolve it.